core/src/main/scala/anorm/Row.scala

```diff
@@ -105,16 +103,9 @@ trait Row {
     }).get // TODO: Safe alternative
 
   // Data per column name
-  private lazy val columnsDictionary: Map[String, Any] = {
-    @annotation.tailrec
-    def loop(meta: List[MetaDataItem], dt: List[Any], r: Map[String, Any]): Map[String, Any] = (meta, dt) match {
-      case (m :: ms, d :: ds) => loop(ms, ds,
-        r + (m.column.qualified.toUpperCase -> d))
-      case _ => r
-    }
-
-    loop(metaData.ms, data, Map.empty)
-  }
+  private lazy val columnsDictionary: Map[String, Any] =
+    (metaData.ms, data).zipped.map((m, v) =>
+      m.column.qualified.toUpperCase -> v).toMap
 
   // Data per column alias
   private lazy val aliasesDictionary: Map[String, Any] = {
```

Comment:
According benchmark (with following data), the tailrec impl is more efficient that a `zipped.foldLeft`.

```scala
val meta = (1 to 10000).map(i => MetaDataItem(ColumnName(s"col$i", Some(s"alias$i")), (i % 2 == 0), "java.lang.String")).toList

val data = (1 to 10000).map(i => s"data-$i").toList
```